src/main/kotlin/uk/gov/justice/digital/hmpps/electronicmonitoringdatastoreapi/service/internal/AuditService.kt

```diff
@@ -0,0 +1,38 @@
+package uk.gov.justice.digital.hmpps.electronicmonitoringdatastoreapi.service.internal
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.springframework.stereotype.Component
+import org.springframework.stereotype.Service
+import software.amazon.awssdk.services.sqs.model.SendMessageRequest
+import uk.gov.justice.digital.hmpps.electronicmonitoringdatastoreapi.model.HmppsAuditEvent
+import uk.gov.justice.hmpps.sqs.HmppsQueue
+import uk.gov.justice.hmpps.sqs.HmppsQueueService
+
+@Service
+@Component
+class AuditService(
+  private val hmppsQueueService: HmppsQueueService,
+  private val objectMapper: ObjectMapper,
+) {
+
+  private val auditQueue by lazy { hmppsQueueService.findByQueueId("audit") as HmppsQueue }
+  private val auditSqsClient by lazy { auditQueue.sqsClient }
+  private val auditQueueUrl by lazy { auditQueue.queueUrl }
+
+  fun createEvent(what: String, detail: String) {
+    auditSqsClient.sendMessage(
+      SendMessageRequest.builder()
+        .queueUrl(auditQueueUrl)
+        .messageBody(
+          objectMapper.writeValueAsString(
+            HmppsAuditEvent(
+              what = what,
+              details = detail,
```

Comment:
hmpps audit event required `details `to be stringified json, sorry it wasn't in the PR i shared as it was found out later.